backend/app/api/endpoints/adapter/tasks.py

```
@@ -320,6 +321,301 @@ async def cancel_task(
     )
 
 
+@router.get("/{task_id}/health")
+async def get_task_health(
+    task_id: int = Depends(with_task_telemetry),
+    current_user: User = Depends(security.get_current_user),
```

Ruff (0.15.6)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)